app/helpers/sidebar_helper.rb

```diff
@@ -33,15 +33,20 @@ def experiment_action_to_link_to(experiment)
   def module_action_to_link_to(my_module)
     case action_name
     when 'results'
-      return results_my_module_url(my_module)
+      results_my_module_url(my_module)
     when 'activities'
-      return activities_my_module_url(my_module)
+      activities_my_module_url(my_module)
     when 'samples'
-      return samples_my_module_url(my_module)
+      samples_my_module_url(my_module)
     when 'archive', 'module_archive', 'experiment_archive'
-      return archive_my_module_url(my_module)
+      archive_my_module_url(my_module)
+    when 'repository' && @repository
```

Comment:
This is not working here it skips this step when the action_name is 'repository'